Fix python bindings for swig 2.0.5.
Somehow swig 2.0.5 produces a long integer instead of an integer in this
situation - I'm not entirely clear on specifics.  But tolerate both as
inputs to svn_stream_read().


--- a/subversion/bindings/swig/core.i
+++ b/subversion/bindings/swig/core.i
@@ -350,12 +350,17 @@
 */
 #ifdef SWIGPYTHON
 %typemap(in) (char *buffer, apr_size_t *len) ($*2_type temp) {
-    if (!PyInt_Check($input)) {
+    if (PyLong_Check($input)) {
+        temp = PyLong_AsLong($input);
+    }
+    else if (PyInt_Check($input)) {
+        temp = PyInt_AsLong($input);
+    }
+    else {
         PyErr_SetString(PyExc_TypeError,
                         "expecting an integer for the buffer size");
         SWIG_fail;
     }
-    temp = PyInt_AsLong($input);
     if (temp < 0) {
         PyErr_SetString(PyExc_ValueError,
                         "buffer size must be a positive integer");
